### ISSUE TITLE
 depguard: create analyzer: couldn't expand $gostd

### ISSUE THREAD

**Issue by piotratlz** [golangci/golangci-lint#3871]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Golangci-lint run on CI  fails with newest version 1.53.0:
```
 Running [/home/runner/golangci-lint-1.53.0-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  level=error msg="[linter] depguard: create analyzer: couldn't expand $gostd: could not read GOROOT directory: open src: no such file or directory\nmust have an Allow and/or Deny package list"
  
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 90ms
```
I was using defaul github actions config:
```yaml
- name: golangci-lint
   uses: golangci/golangci-lint-action@v3
```
I do not get this issue on version 1.52.2.

I am aware that this is related to github action, however the root cause of the issue is version bump on CI.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
1.53.0
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```

</details>

**Comment by ldez** [comment]
duplicate of #3862, #3870

**Comment by caveman280** [comment]
I hope this PR will help fix users of GHA: golangci/golangci-lint#3872
